### DISCUSSION TITLE
GraphRAG visualization and UI

### DISCUSSION THREAD

**Discussion by natoverse** [microsoft/graphrag#697] [Ideas]
Placeholder discussion for visualization and UI, which community members have commonly posted examples and solutions for

**Comment by natoverse** [comment]
https://github.com/microsoft/graphrag/issues/566

**Reply by natoverse** [reply]
https://github.com/microsoft/graphrag/issues/689

**Reply by natoverse** [reply]
https://github.com/microsoft/graphrag/discussions/427

**Reply by natoverse** [reply]
https://github.com/microsoft/graphrag/discussions/576

**Comment by natoverse** [comment]
Some notes I've put elsewhere:
A couple of suggestions:

1. When running your index, you can set GRAPHRAG_SNAPSHOT_GRAPHML=True to output graphml files that can be visualized in a desktop tool like [Gephi](https://gephi.org/).
2. We have [config](https://microsoft.github.io/graphrag/posts/config/env_vars/) for node2vec and umap, which will embed the graph and then do dimensionality reduction so each node is assigned an x/y coordinate in the output. This is useful for plotting programs, and will locate the nodes near their community clusters.

**Reply by OMGLASERSPEWPEWPEW** [reply]
Sorry I'm a newbie developer.  When I look at the settings.yaml, I can't find that setting.  Is it somewhere else? I installed graphrag via pip install and it's the latest version 0,2,2

**Reply by BobbyAxelrods** [reply]
> Sorry I'm a newbie developer. When I look at the settings.yaml, I can't find that setting. Is it somewhere else? I installed graphrag via pip install and it's the latest version 0,2,2

Its there, 


1. Enable the snapshot of graphml = true & umap = true . Good to go 
![image](https://github.com/user-attachments/assets/bcc46aef-e1e0-4f27-a97f-4a47eadf265f)

Other option you may upload artificat to viz the network 
https://github.com/microsoft/graphrag/discussions/884 

Hope this helps

**Comment by KylinMountain** [comment]
@natoverse  so when enable the umap and node2vec，what is the output file name? And how can generate a 3D graphml? Any suggestion is appreciated thank you.

**Reply by natoverse** [reply]
The `create_final_nodes.parquet` has an x and y for each node. These are set to 0 by default, but if you enable umap and node2vec they will be populated with a coordinate that you can plot. We don't currently expose the ability to compute a z position for 3D plotting.

**Comment by KylinMountain** [comment]
besides i have another question about is there any weight to metric the entity or relationship importance? Thansk

**Reply by natoverse** [reply]
The `rank` field for entity nodes matches the degree - i.e., we simply use the node degree as the rank. For the relationship `rank` we sum the degrees of the two connected nodes. So both of those consider the connectedness of the nodes for assigning a weight.

**Reply by KylinMountain** [reply]
thank you very much, this is very helpful to me. What rank the really meanings of the entity? Looks like we didn’t add count field to the entity, is this also important in graph knowledge?

**Reply by natoverse** [reply]
The rank is the count of connected edges, which can be computed from the edge list without storing any count on the entity.
